lib/job.js

```diff
@@ -174,8 +204,11 @@ class Job extends BaseModel {
                     .then(() => removeBuilds());
             }));
 
-        // Remove the job
-        return removeBuilds().then(() => super.remove());
+        // Remove builds
+        return removeBuilds()
+            // Remove periodic job
+            .then(() => this[executor].stopPeriodic({ jobId: this.id }))
```

Comment:
shouldn't it check whether it's a periodic job prior to invoking `stopPeriodic`

Reply:
I was thinking the check should be done in `executor-queue`, since it can simply check if there is a record in redis, but I can also do an extra check here.